watertap/examples/case_studies/municipal_treatment/municipal_treatment.py

```diff
@@ -264,28 +220,74 @@ def display_results(m):
     unit_list = ['feed', 'intake_pump', 'coag_and_floc',
                  'sedimentation', 'ozonation', 'gravity_basin',
                  'gac', 'backwash_pump', 'uv', 'anion_exchange',
-                 'chlorination', 'storage', 'recharge_pump']
+                 'chlorination', 'storage', 'recharge_pump', 'product']
 
     for u in unit_list:
-        if hasattr(m.fs, u):
-            unit = getattr(m.fs, u)
-            unit.report()
+        m.fs.component(u).report()
 
 
 def add_costing(m):
-    # TODO: add costing
-    pass
+    m.fs.costing = ZeroOrderCosting()
+    # typing aid
+    costing_kwargs = {"default": {"flowsheet_costing_block": m.fs.costing}}
+    m.fs.intake_pump.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.coag_and_floc.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.sedimentation.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.ozonation.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.gravity_basin.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.gac.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.backwash_pump.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.uv.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.anion_exchange.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.chlorination.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.storage.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.recharge_pump.costing = UnitModelCostingBlock(**costing_kwargs)
+
+    m.fs.costing.cost_process()
+    m.fs.costing.add_electricity_intensity(m.fs.product.properties[0].flow_vol)
+    m.fs.costing.add_LCOW(m.fs.product.properties[0].flow_vol)
 
 
 def initialize_costing(m):
-    # TODO: initialize costing
-    pass
+    m.fs.costing.initialize()
 
 
 def display_costing(m):
-     # TODO: display costing
-    pass
+
+    m.fs.costing.total_capital_cost.display()
+    m.fs.costing.total_operating_cost.display()
+    m.fs.costing.LCOW.display()
+
+    print("\nUnit Capital Costs\n")
+    for u in m.fs.costing._registered_unit_costing:
+        print(u.name,
+              " :   ",
+              value(pyunits.convert(u.capital_cost,
+                                    to_units=pyunits.USD_2018)))
+    for u in m.fs.costing._registered_unit_costing:
+        print(u.name,
+              " :   ",
+              value(pyunits.convert(u.capital_cost,
+                                    to_units=pyunits.USD_2018)))
```

Comment:
duplicate code?

Reply:
Yeah -- thanks. I didn't notice because I copied it from the seawater_RO_desalination flowsheet. I've now fixed it in both places.